docs/arcadia-users-group/20240206-intro-to-formatting-and-linting/lesson.md

```diff
@@ -0,0 +1,425 @@
+
+
+### Setting up `black`
+To install `black` in our conda environment, run:
+```bash
+mamba install -y black
+```
+
+Next, let's add some poorly formatted code to our toy project. Add the following code (from our previous example) to `main.py`:
+```python
+def calculate_total_income(gross_wages, taxable_interest, total_dividends, qualified_dividends, other_income):
+    total_income = gross_wages + taxable_interest + (total_dividends - qualified_dividends) + other_income
+    return total_income
+```
+Now, from the command line, run:
+```bash
+black .
+```
+This command tells `black` to reformat all the Python files in the current directory (the lone dot in the command is the relative path to the current directory). If you look at `main.py` now, you'll see that the code has been reformatted according to `black`'s rules.
+
+Now, try simplifying the calculation of `total_income` by removing the dividends from the sum, then run `black` again. You should see that, because the `total_income` line is shorter, it can fit on one line, so `black` automatically converts it back to a single line.
+
+There is a succinct overview of `black`'s formatting rules in [its documentation](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html). While it's good to be familiar with these rules, there is no need to memorize them. Indeed, the purpose of using a formatting tool like `black` is to eliminate the need to manually write your code according to any particular formatting conventions or even to think about formatting at all. 
+
+
+## Style guides
+Although formatting conventions help constrain the very low-level aspects of how code is written, additional rules and conventions are required to ensure that code is both readable and understandable. These conventions are usually expressed in the form of style guides. Style guides define rules, conventions, and guidelines to which all code in a codebase should adhere. They concern both both low-level questions like how to name variable and functions, as well as higher-level considerations like documentation standards and some aspects of how code is organized or structured.
+
+### PEP8 and Python style guides
+Many language have an official style guide. For Python, the official style guide is called [PEP8](https://peps.python.org/pep-0008/#introduction). In addition, Google's [Python style guide](https://google.github.io/styleguide/pyguide.html) is an important extension of PEP8 that defines additional standards, especially around comments and documentation. Most major Python projects follow at least PEP8, and we should strive to follow it at Arcadia as well. This is true no matter the scale of the project; it is a good practice to follow PEP8 even for small scripts and one-off analyses. As projects grow, it is usually wise to start imposing additional style conventions from Google' style guide on top of PEP8.
```

Comment:
Ah good point! I'll add that. I think it'll be much more effective to provide a specific motivation, rather than a vague "do this or else" prescription.